code/game/objects/items/devices/scanners.dm

```diff
@@ -245,6 +245,14 @@ REAGENT SCANNER
 		if(H.heart_attack)
 			user.show_message("<span class='userdanger'>Subject suffering from heart attack: Apply defibrillator immediately.</span>")
 		user.show_message("\blue Subject's pulse: <font color='[H.pulse == PULSE_THREADY || H.pulse == PULSE_NONE ? "red" : "blue"]'>[H.get_pulse(GETPULSE_TOOL)] bpm.</font>")
+		var/implant_detect
+		for(var/obj/item/organ/internal/cyberimp/CI in H.internal_organs)
+			if(CI.status == ORGAN_ROBOT && !(H.species && H.species.name == "Machine"))

```

Comment:
Perhaps implant analysis could be left to the cyborg analyzer, which already handles the status of robotic organs, while the mere presence or absence of robotic implants could be within the upgraded health analyzer's domain. I'm not sure why you'd want to exclude IPCs from having internal implants detected, though.